lib/compilers/ppci.js

```diff
@@ -0,0 +1,29 @@
+
+const Compile = require('../base-compiler'),
+    utils = require('../utils');
+
+function compilePpci(info, env) {
+    const compiler = new Compile(info, env);
+
+    compiler.runCompiler = function (compiler, options, inputFilename, execOptions) {
+        if (!execOptions) {
+            execOptions = compiler.getDefaultExecOptions();
+        }
+
+        compiler = 'python3';
+        python_path = '/opt/compiler-explorer/ppci-0.5.5/';
```

Comment:
`python_path` is undefined. You are missing a `const` right before the name

Reply:
Also, note that the hardcoded compiler path is probably not a good idea. You have `complier.compiler.exe` available that will give you the exe you specified in the properties file. That means you probably want to do this the other way around. Compile a program by executing python, and passing it the necessary options. First the ones needed to run the compiler, and then the ones needed from the compiler to compile the input. I hope ths makes some sense. Feel free to ask if you have any doubts

Reply:
This is indeed the idea. Run python and pass it extra options to locate the compiler. But the problem remains the path that must be set. Another option might be to create a bash script in the compiler explorer image which does this extra actions. Then there is no need for an extra compiler type.